ts/private/ts_project.bzl

```diff
@@ -241,18 +243,20 @@ This is an error because Bazel does not run actions unless their outputs are nee
     stdout_file = ""
 
     if ctx.attr.no_emit:
-        # Validation actions still need to produce some output, so we output the stdout
-        # to a .validation file that ends up in the _validation output group.
-        validation_output = ctx.actions.declare_file(ctx.attr.name + ".validation")
-        validation_outs.append(validation_output)
+        # The type-checking action still need to produce some output, so we output the stdout
+        # to a .typecheck file that ends up in the typecheck output group.
+        typecheck_output = ctx.actions.declare_file(ctx.attr.name + ".typecheck")
+        typecheck_outs.append(typecheck_output)
 
-        outputs.append(validation_output)
-        stdout_file = validation_output.path
+        outputs.append(typecheck_output)
+        stdout_file = typecheck_output.path
 
         if supports_workers:
-            arguments.add_all(["--bazelValidationFile", validation_output.short_path])
+            arguments.add_all(["--bazelValidationFile", typecheck_output.short_path])
```

Comment:
Not: you can use the 2-param overload of add, no need to construct an array and call add_all

Reply:
Nit*